tsdb/db.go

```diff
@@ -538,6 +538,17 @@ func Open(dir string, l log.Logger, r prometheus.Registerer, opts *Options) (db
 			segmentSize = opts.WALSegmentSize
 		}
 		wlog, err = wal.NewSize(l, r, filepath.Join(dir, "wal"), segmentSize, opts.WALCompression)
+		if err != nil && errors.Cause(err).Error() == wal.ErrorUnsequentialSegments {
+			level.Warn(db.logger).Log("msg", "encountered WAL segment sequence error, attempting repair", "err", err)
+			level.Warn(db.logger).Log("msg", "deleting unsequential segments")
```

Comment:
This message is not clear as to which segments are being deleted. Before or after the gap.